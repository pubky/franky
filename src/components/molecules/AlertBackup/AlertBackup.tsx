import * as Atoms from '@/atoms';
import * as Libs from '@/libs';
import * as Molecules from '@/molecules';

export const AlertBackup = () => {
  return (
    <Atoms.Container className="px-6 py-3 bg-brand rounded-lg flex flex-row items-center gap-3">
      <Atoms.Container className="flex flex-row flex-1 gap-3 items-center">
        <Libs.TriangleAlert className="h-4 w-4 font-bold text-primary-foreground" />
        <Atoms.Typography size="sm" className="font-bold text-primary-foreground">
          Back up now to avoid losing your account!
        </Atoms.Typography>
      </Atoms.Container>
      <Molecules.DialogBackup />
      <Molecules.DialogConfirmBackup />
    </Atoms.Container>
  );
};
