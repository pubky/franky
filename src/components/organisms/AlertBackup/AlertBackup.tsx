'use client';

import { useState, useEffect } from 'react';
import * as Atoms from '@/atoms';
import * as Libs from '@/libs';
import * as Core from '@/core';
import * as Organisms from '@/organisms';

/**
 * AlertBackup
 *
 * Self-contained component that manages its own visibility state.
 * Shows an alert when user has a secret key that needs to be backed up.
 * No props needed - manages its own state internally.
 */
export const AlertBackup = () => {
  const { secretKey } = Core.useOnboardingStore();
  const [showAlert, setShowAlert] = useState(false);

  useEffect(() => {
    if (secretKey) {
      setShowAlert(true);
    }
  }, [secretKey]);

  const handleDismiss = () => {
    setShowAlert(false);
  };

  // Don't show alert if there's no secretKey or it was dismissed
  if (!secretKey || !showAlert) {
    return null;
  }

  return (
    <Atoms.Container className="px-6 py-3 bg-brand rounded-lg flex-row items-center gap-3">
      <Atoms.Container className="flex-row flex-1 gap-3 items-center">
        <Libs.TriangleAlert className="h-4 w-4 font-bold text-primary-foreground" />
        <Atoms.Typography size="sm" className="font-bold text-primary-foreground whitespace-nowrap">
          Back up now<span className="hidden md:inline"> to avoid losing your account!</span>
        </Atoms.Typography>
      </Atoms.Container>
      <Organisms.DialogBackup />
      <Organisms.DialogConfirmBackup onConfirm={handleDismiss} />
    </Atoms.Container>
  );
};
