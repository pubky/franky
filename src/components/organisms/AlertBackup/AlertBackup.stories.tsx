import type { Meta, StoryObj } from '@storybook/nextjs-vite';
import { AlertBackup } from './AlertBackup';

const meta = {
  title: 'Molecules/AlertBackup',
  component: AlertBackup,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
  argTypes: {
    onDismiss: { action: 'dismissed' },
  },
} satisfies Meta<typeof AlertBackup>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {},
};

export const WithDismissHandler: Story = {
  args: {
    onDismiss: () => console.log('Dismissed'),
  },
};
