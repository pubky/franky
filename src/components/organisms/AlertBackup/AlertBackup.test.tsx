import { describe, it, expect, vi } from 'vitest';
import { render, screen } from '@testing-library/react';
import { AlertBackup } from './AlertBackup';

// Mock Core module
vi.mock('@/core', () => ({
  useOnboardingStore: vi.fn(() => ({
    secretKey: 'test-secret-key',
  })),
}));

// Mock atoms
vi.mock('@/atoms', () => ({
  Container: ({ children, className }: { children: React.ReactNode; className?: string }) => (
    <div data-testid="container" className={className}>
      {children}
    </div>
  ),
  Typography: ({ children, size, className }: { children: React.ReactNode; size?: string; className?: string }) => (
    <p data-testid="typography" data-size={size} className={className}>
      {children}
    </p>
  ),
  Dialog: vi.fn(({ children }: { children: React.ReactNode }) => <div data-testid="dialog">{children}</div>),
  DialogTrigger: vi.fn(({ children }: { children: React.ReactNode }) => (
    <div data-testid="dialog-trigger">{children}</div>
  )),
  DialogClose: vi.fn(({ children }: { children: React.ReactNode }) => <div data-testid="dialog-close">{children}</div>),
  DialogContent: ({ children, className }: { children: React.ReactNode; className?: string }) => (
    <div data-testid="dialog-content" className={className}>
      {children}
    </div>
  ),
  DialogHeader: ({ children }: { children: React.ReactNode }) => <div data-testid="dialog-header">{children}</div>,
  DialogTitle: ({ children }: { children: React.ReactNode }) => <h2 data-testid="dialog-title">{children}</h2>,
  DialogDescription: ({ children, className }: { children: React.ReactNode; className?: string }) => (
    <div data-testid="dialog-description" className={className}>
      {children}
    </div>
  ),
  Button: ({
    children,
    variant,
    className,
    onClick,
  }: {
    children: React.ReactNode;
    variant?: string;
    className?: string;
    onClick?: () => void;
  }) => (
    <button data-testid="button" data-variant={variant} className={className} onClick={onClick}>
      {children}
    </button>
  ),
  Card: ({ children, className }: { children: React.ReactNode; className?: string }) => (
    <div data-testid="card" className={className}>
      {children}
    </div>
  ),
  Label: ({ children, htmlFor, className }: { children: React.ReactNode; htmlFor?: string; className?: string }) => (
    <label data-testid="label" htmlFor={htmlFor} className={className}>
      {children}
    </label>
  ),
  Input: ({
    type,
    id,
    placeholder,
    value,
    onChange,
    className,
  }: {
    type?: string;
    id?: string;
    placeholder?: string;
    value?: string;
    onChange?: (e: React.ChangeEvent<HTMLInputElement>) => void;
    className?: string;
  }) => (
    <input
      data-testid="input"
      type={type}
      id={id}
      placeholder={placeholder}
      value={value}
      onChange={onChange}
      className={className}
    />
  ),
}));

// Mock libs
vi.mock('@/libs', () => ({
  TriangleAlert: ({ className }: { className?: string }) => (
    <div data-testid="triangle-alert" className={className}>
      TriangleAlert
    </div>
  ),
  ArrowLeft: ({ className }: { className?: string }) => (
    <div data-testid="arrow-left" className={className}>
      ArrowLeft
    </div>
  ),
  Download: ({ className }: { className?: string }) => (
    <div data-testid="download" className={className}>
      Download
    </div>
  ),
  Eye: ({ className }: { className?: string }) => (
    <div data-testid="eye" className={className}>
      Eye
    </div>
  ),
  EyeOff: ({ className }: { className?: string }) => (
    <div data-testid="eye-off" className={className}>
      EyeOff
    </div>
  ),
  Copy: ({ className }: { className?: string }) => (
    <div data-testid="copy" className={className}>
      Copy
    </div>
  ),
  Check: ({ className }: { className?: string }) => (
    <div data-testid="check" className={className}>
      Check
    </div>
  ),
  ShieldCheck: ({ className }: { className?: string }) => (
    <div data-testid="shield-check" className={className}>
      ShieldCheck
    </div>
  ),
}));

// Mock molecules
vi.mock('@/molecules', () => ({
  DialogBackupPhrase: vi.fn(() => <div data-testid="dialog-backup-phrase">DialogBackupPhrase</div>),
  DialogExport: vi.fn(() => <div data-testid="dialog-export">DialogExport</div>),
}));

// Mock organisms
vi.mock('@/organisms', () => ({
  DialogBackup: vi.fn(() => <div data-testid="dialog-backup">DialogBackup</div>),
  DialogConfirmBackup: vi.fn(() => <div data-testid="dialog-confirm-backup">DialogConfirmBackup</div>),
  DialogBackupEncrypted: vi.fn(() => <div data-testid="dialog-backup-encrypted">DialogBackupEncrypted</div>),
}));

describe('AlertBackup', () => {
  it('renders all required elements', () => {
    render(<AlertBackup />);

    expect(screen.getByTestId('triangle-alert')).toBeInTheDocument();
    expect(screen.getByText('Back up now')).toBeInTheDocument();
    expect(screen.getByText('to avoid losing your account!')).toBeInTheDocument();
    expect(screen.getByTestId('dialog-backup')).toBeInTheDocument();
    expect(screen.getByTestId('dialog-confirm-backup')).toBeInTheDocument();
  });
});

describe('AlertBackup - Snapshot', () => {
  it('matches snapshot for default AlertBackup', () => {
    const { container } = render(<AlertBackup />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
