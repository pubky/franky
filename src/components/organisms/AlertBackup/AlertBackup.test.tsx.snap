// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AlertBackup - Snapshot > matches snapshot for default AlertBackup 1`] = `
<div
  class="px-6 py-3 bg-brand rounded-lg flex-row items-center gap-3"
  data-testid="container"
>
  <div
    class="flex-row flex-1 gap-3 items-center"
    data-testid="container"
  >
    <svg
      aria-hidden="true"
      class="lucide lucide-triangle-alert h-4 w-4 font-bold text-primary-foreground"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="m21.73 18-8-14a2 2 0 0 0-3.48 0l-8 14A2 2 0 0 0 4 21h16a2 2 0 0 0 1.73-3"
      />
      <path
        d="M12 9v4"
      />
      <path
        d="M12 17h.01"
      />
    </svg>
    <p
      class="font-bold text-primary-foreground whitespace-nowrap"
      data-size="sm"
      data-testid="typography"
    >
      Back up now
      <span
        class="hidden md:inline"
      >
         to avoid losing your account!
      </span>
    </p>
  </div>
  <div
    data-testid="dialog-backup"
  >
    DialogBackup
  </div>
  <div
    data-testid="dialog-confirm-backup"
  >
    DialogConfirmBackup
  </div>
</div>
`;
